nomad/server.go

```diff
@@ -1430,17 +1440,165 @@ func (s *Server) setupSerf(conf *serf.Config, ch chan serf.Event, path string) (
 	return serf.Create(conf)
 }
 
+// shouldReloadSchedulers checks the new config to determine if the scheduler worker pool
+// needs to be updated. If so, returns true and a pointer to a populated SchedulerWorkerPoolArgs
+func shouldReloadSchedulers(s *Server, newPoolArgs *SchedulerWorkerPoolArgs) (bool, *SchedulerWorkerPoolArgs) {
```

Comment:
In retrospect we probably could have unconditionally drained and replaced the worker pool on reload; the API call is gated by ACLs so the risk of an operator DoS'ing their own scheduler seems low. And then we could come back in later to do this more clever logic in a future PR. But this is pretty nice and enables scale up/down in the future (as per your [`// TODO`](https://github.com/hashicorp/nomad/pull/11593/files#diff-ae3d4d8a6bea47048b58ab5b1bcec43a443bafb1cae482fc59793ab8818aeb1aR1573) remark below).